picard/formats/__init__.py

```diff
@@ -39,17 +40,47 @@ def supported_formats():
     return formats
 
 
+def guess_format(filename, options=_formats):
+    """Select the best matching file type amongst supported formats."""
+    results = []
+    with file(filename, "rb") as fileobj:
```

Comment:
From https://docs.python.org/2/library/functions.html#file:

> **file(name[, mode[, buffering]])**
>Constructor function for the file type, described further in section File Objects. The constructor’s arguments are the same as those of the open() built-in function described below.
>
>When opening a file, it’s preferable to use open() instead of invoking this constructor directly. file is more suited to type testing (for example, writing isinstance(f, file)).

Of course there are other open methods used in this code other than the native Python file open() - so using open instead of file might reduce readability rather than improve it.